[naga] Centralize naming of anonymous entry point return types.

jimblandy opened the pull request:
When an entry point's return type is anonymous, have `naga::proc::Namer` assign it a name based on its shader stage.

Remove bespoke logic for this from the WGSL backend.

Fixes #7263.

ErichDonGubler commented on naga/tests/in/glsl/quad_glsl.frag:
```diff
@@ -6,6 +6,10 @@ layout(set = 0, binding = 1) uniform sampler u_sampler;
 #endif
 layout(location = 0) out vec4 o_color;
 
+struct FragmentOutput {
+  float x;
+};
+
```
nitpick: I think it'd be better to have self-contained test(s) for the coverage specifically, rather than piggybacking in a way where coverage is not clearly lost if the test gets changed for other reasons.

jimblandy replied:
I feel the same way - I just wasn't sure how people would feel. And for mysterious reasons I just happen to have a simple test case handy!

ErichDonGubler reviewed (approved):
LGTM! I have a nit, but it can wait for another PR.

jimblandy committed:
[naga] Centralize naming of anonymous entry point return types.

When an entry point's return type is anonymous, have
`naga::proc::Namer` assign it a name based on its shader stage.

Remove bespoke logic for this from the WGSL backend.

Fixes #7263.
